src/main/java/seedu/foodrem/logic/commands/tagcommands/RenameTagCommand.java

```diff
@@ -0,0 +1,68 @@
+package seedu.foodrem.logic.commands.tagcommands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.foodrem.logic.parser.CliSyntax.PREFIX_NAME;
+
+import seedu.foodrem.logic.commands.Command;
+import seedu.foodrem.logic.commands.CommandResult;
+import seedu.foodrem.logic.commands.exceptions.CommandException;
+import seedu.foodrem.model.Model;
+import seedu.foodrem.model.tag.Tag;
+
+/**
+ * Renames an existing tag in FoodRem.
+ */
+public class RenameTagCommand extends Command {
+
+    public static final String COMMAND_WORD = "renametag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Renames an existing tag in FoodRem. "
+            + "Parameters: "
+            + PREFIX_NAME + "ORIGINAL_TAG_NAME "
+            + PREFIX_NAME + "NEW_TAG_NAME "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "Vegetables "
+            + PREFIX_NAME + "fruits";
+
+    public static final String MESSAGE_RENAME_TAG_SUCCESS = "Renamed tag: %1$s";
+    public static final String MESSAGE_TAG_NOT_FOUND = "This tag does not exist in the FoodRem.";
+    public static final String MESSAGE_DUPLICATE_TAG_NAME = "This tag name already exists in the FoodRem.";
+
+    private final Tag originalTag;
+    private final Tag renamedTag;
+
+    /**
+     * Creates an AddTagCommand to add the specified {@code Tag}
+     */
+    public RenameTagCommand(Tag originalTag, Tag renamedTag) {
+        requireNonNull(originalTag);
+        requireNonNull(renamedTag);
+        this.originalTag = originalTag;
+        this.renamedTag = renamedTag;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (!model.hasTag(originalTag)) {
+            throw new CommandException(MESSAGE_TAG_NOT_FOUND);
+        }
+
+        if (!originalTag.equals(renamedTag) && model.hasTag(renamedTag)) {
```

Comment:
Question:
For this condition, would it be clearer to write 
If originaltag.equals(renamedtag) || model.hasTag(renamedTag)

Reply:
Hmm, I just following the same condition in the EditCommand.

However, I think this is slightly clearer since this explicitly states that it is checking whether the tagList contains another tag that's the same as renamedTag but not the originalTag.

Reply:
Ok sure